src/Pyrexx/ZUGFeRD/Model/AllowanceCharge.php

```diff
@@ -27,6 +28,22 @@ class AllowanceCharge
      */
     private $actualAmount;
 
+    /**
+     * @var string
+     * @JMS\Type("string")
+     * @JMS\XmlElement(cdata=false, namespace="urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:12")
+     * @JMS\SerializedName("Reason")
+     */
+    private $reason;
+
+    /**
+     * @var TradeTax
```

Comment:
should not we introduce a new type CategoryTradeTax?
CategoryTradeTax has just three properties and if we use TradeTax user can be confused or accidentally set properties that should not be set.

Reply:
Hi, according to the scheme **CategoryTradeTax** is the name of the xml element, but the type of the element is TradeTaxType, [link to the ZUGFeRD scheme](https://github.com/Pyrexx/zugferd-php/blob/master/src/Pyrexx/ZUGFeRD/Assets/Schema/ZUGFeRD1p0_urn_un_unece_uncefact_data_standard_ReusableAggregateBusinessInformationEntity_12.xsd#L231) These are all fields that TradeTaxType could use: [TradeTaxType children](https://github.com/Pyrexx/zugferd-php/blob/master/src/Pyrexx/ZUGFeRD/Assets/Schema/ZUGFeRD1p0_urn_un_unece_uncefact_data_standard_ReusableAggregateBusinessInformationEntity_12.xsd#L334) If you decide that a new model is a better way I will create it.

Reply:
you are right it's TradeTaxType. I just checked that in https://konik.io/ZUGFeRD-Spezifikation/Das-ZUGFeRD-Format_1p0_technische_Dokumentation.pdf only three fields from the TradeTaxType are used when for the tax is inside SpecifiedTradeAllowanceCharge tag (page 31 and page 72)
But I agree, we don't need an extra type for that.